src/services/shareService.ts

```diff
@@ -0,0 +1,184 @@
+        
+        const card = stringToObjectId(cardId);
+
+        if (!card) {
+            return { error: 'card not found' };
+        }
+
+        // TODO: Get USER_ID from session
+        const user = stringToObjectId('123456789012');
+
+        if (!user) {
+            return { error: 'user not found' };
+        }
+        const now = new Date();
+        const valid_until = now.setTime(now.getTime()+(validHours*60*60*1000));
+
+        const sharing: ISharing = await SharingModel.create({ user, card, valid_until });
+
+        return {
+            sharing,
+        };
+    } catch (error) {
+        return { error };
+    }
+}
+
+export async function getSharing(id: string) {
+    try {
+        await connectDB();
+        // TODO: Get USER_ID from session
+        const user = stringToObjectId('123456789012');
+
+        if (!user) {
+            return { error: 'user not found' };
+        }
+
+        const parsedId = stringToObjectId(id);
+
+        if (!parsedId) {
+            return { error: 'Sharing not found' };
+        }
+        const sharing = await SharingModel.findOne<ISharing>({ _id: parsedId, user });
+        if (sharing) {
+            return {
+                sharing,
+            };
+        } else {
+            return { error: 'Sharing not found' };
+        }
+    } catch (error) {
+        return { error };
+    }
+}
+
+export async function fetchSharing(id: string) {
+    try {
+        await connectDB();
+
+        const parsedId = stringToObjectId(id);
+
+        if (!parsedId) {
+            return { error: 'Sharing not found' };
+        }
+
+        const sharing = await SharingModel.findOne<ISharing>({ _id: parsedId, valid_until: { $gte: new Date() }, isComplete: false });
+        if (sharing) {
+            return {
+                sharing,
+            };
+        } else {
+            return { error: 'Sharing not found' };
+        }
+    } catch (error) {
+        return { error };
+    }
+}
+
+export async function executeSharing(id: string) {
+    try {
+        await connectDB();
+        // TODO: Get USER_ID from session
+        const user = stringToObjectId('123456789012');
+
+        if (!user) {
+            return { error: 'user not found' };
+        }
+
+        const { sharing } = await fetchSharing(id);
+
+        if (!sharing) {
+            return { error: 'Sharing not found' };
+        }
+        const { card } : {card:ICard} = await sharing.populate('card');
+
+        const { types } = await getTypes();
+        if (!types) {
+            return { error: 'Unable to copy card' };
+        }
+        const defaultType = types.filter(t => t.isDefault)[0];
+        if (!defaultType) {
+            return { error: 'Unable to copy card' };
+        }
+        const newCard = await createCard(new CardAdaptor({image_front: card.image_front, image_back: card.image_back, name: card.name, type_id: defaultType._id!, number: card.number, note: card.note}));
+        if (newCard) {
```

Comment:
createCard returns an object of { card } or { error }.

Reply:
Fixed: https://github.com/tsoibet/kaado/pull/14/commits/ab2246d599ef556371ce94663f0958bd0ee15837